### PR TITLE
Add default keybinding for opening dropdown

### LINKED ISSUE #2181: New Tab Drop-down should be keyboard accessible
# Description of the new feature/enhancement
Users should be able to display the "new tab drop-down" using a key chord.

I'd prefer <kbd>CTRL</kbd><kbd>SHIFT</kbd><kbd>SPACE</kbd> but I am sure others would prefer a different chord, so this chord-action should key-mappable.

### PR DESCRIPTION
**Pull request by cinnamon-msft**
## Summary of the Pull Request
Adds default keybinding of <kbd>Ctrl</kbd><kbd>Shift</kbd><kbd>Space</kbd> to open the dropdown menu.

 
## PR Checklist
* [x] Closes #2181
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [x] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Detailed Description of the Pull Request / Additional comments
<kbd>Ctrl</kbd><kbd>Shift</kbd><kbd>Space</kbd> will not close the dropdown menu if it has been opened. Do we want this functionality or is <kbd>ESC</kbd> sufficient?

### CODE REVIEW

File: src/cascadia/TerminalApp/AppKeyBindings.idl [hunk 1]
```diff
@@ -17,6 +17,7 @@ namespace TerminalApp
         CopyTextWithoutNewlines,
         PasteText,
         NewTab,
+        Dropdown,
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/AppKeyBindings.idl]: Can we rename this (and everything else) `ToggleNewTabDropdown`? We may have multiple dropdown-like things in the future, I don't want to back us into a corner by naming this an unqualified "dropdown".

File: src/cascadia/TerminalApp/App.cpp [hunk 2]
```diff
@@ -405,6 +405,13 @@ namespace winrt::TerminalApp::implementation
         _newTabButton.Flyout(newTabFlyout);
     }
 
+    void App::_OpenDropdown()
+    {
+        Controls::Primitives::FlyoutShowOptions options{};
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/App.cpp]: What happens when the dropdown is already visible?

File: src/cascadia/TerminalApp/App.cpp [hunk 3]
```diff
@@ -405,6 +405,13 @@ namespace winrt::TerminalApp::implementation
         _newTabButton.Flyout(newTabFlyout);
     }
 
+    void App::_OpenDropdown()
+    {
+        Controls::Primitives::FlyoutShowOptions options{};
+        options.Placement(Controls::Primitives::FlyoutPlacementMode::BottomEdgeAlignedLeft);
+        _newTabButton.Flyout().ShowAt(_newTabButton, options);
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/App.cpp]: Does this move UIA focus into the new tab dropdown? Should it? cc @carlos-zamora

File: src/cascadia/TerminalApp/App.cpp [hunk 4]
```diff
@@ -405,6 +405,13 @@ namespace winrt::TerminalApp::implementation
         _newTabButton.Flyout(newTabFlyout);
     }
 
+    void App::_OpenDropdown()
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/App.cpp]: This should have a doc comment

File: src/cascadia/TerminalApp/App.cpp (hunk 2 above)
Comment (cinnamon-msft) [reply src/cascadia/TerminalApp/App.cpp]: The keybinding is ignored and the <kbd>Space</kbd> key will open the first profile in the list

Comment (zadjii-msft) [reply src/cascadia/TerminalApp/App.cpp]: Oh that's an actually good behavior.

File: src/cascadia/TerminalApp/AppKeyBindings.idl (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/AppKeyBindings.idl]: Okay let's not call it `Toggle` but the rest of my comment stands

File: src/cascadia/TerminalApp/App.cpp (hunk 3 above)
Comment (carlos-zamora) [reply src/cascadia/TerminalApp/App.cpp]: Good news: Xaml handles all of that for us as we change focus with the keyboard
"Bad" news: navigation between the flyout and its parent isn't great (but that's an existing issue). That'll get fixed in #2102 by me

File: src/cascadia/TerminalApp/AppKeyBindings.idl (hunk 1 above)
Comment (cinnamon-msft) [reply src/cascadia/TerminalApp/AppKeyBindings.idl]: Is `OpenNewTabDropdown` good?

Comment (zadjii-msft) [reply src/cascadia/TerminalApp/AppKeyBindings.idl]: Seems _perfect_ to me :)

File: src/cascadia/TerminalApp/App.cpp [hunk 5]
```diff
@@ -405,6 +405,18 @@ namespace winrt::TerminalApp::implementation
         _newTabButton.Flyout(newTabFlyout);
     }
 
+    // Function Description:
+    // Called when the openNewTabDropdown keybinding is used.
+    // Adds the flyout show option to left-align the dropdown with the split button.
+    // Shows the dropdown flyout.
+
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/App.cpp]: nit: extra newline here

**Comment by DHowett-MSFT** [comment]
If this becomes the generic "menu" that we use for all user-configurable things it will no longer be just the "newTabDropdown". I'm not sure I carE? I dunno, the name seems _too_ specific now.

**Comment by miniksa** [comment]
I just merged master to this branch to unblock the static analysis build.

**Pull request merged**